test/InjectorTest.php

```diff
@@ -341,4 +342,127 @@ public function testComplexDeepDependencyConfiguration()
         $this->assertSame($expected1, $result->result->result->optionalResult);
         $this->assertSame($expected2, $result->result2->result->optionalResult);
     }
+
+    public function testCreateInstanceWithoutUnknownClassThrowsException()
+    {
+        $this->expectException(Exception\ClassNotFoundException::class);
+        (new Injector())->create('Unknown.Alias.Should.Fail');
+    }
+
+    public function testKnownButInexistentClassThrowsException()
+    {
+        $definition = $this->getMockBuilder(DefinitionInterface::class)
+            ->getMockForAbstractClass();
+
+        $definition->expects($this->any())
+            ->method('hasClass')
+            ->willReturn(true);
+
+        $this->expectException(Exception\ClassNotFoundException::class);
+        (new Injector(null, null, $definition))->create('ZendTest\Di\TestAsset\No\Such\Class');
+    }
+
+    public function provideUnexpectedResolverValues()
+    {
+        return [
+            'string' => [ 'string value' ],
+            'bool' => [ true ],
+            'null' => [ null ],
+            'object' => [ new stdClass() ]
```

Comment:
Note to self: check for trailing commas on final list items, alignment, and whitespace around braces.